src/libs/ReportActionsUtils.ts

```diff
@@ -997,22 +1000,27 @@ function getOneTransactionThreadReportID(reportID: string, reportActions: OnyxEn
                 // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
                 (isMessageDeleted(action) && action.childVisibleActionCount) ||
                 (action.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE && (isOffline ?? isNetworkOffline)))
-        );
-    });
+        ) {
+            iouRequestActions.push(action);
+        }
+    }
 
     // If we don't have any IOU request actions, or we have more than one IOU request actions, this isn't a oneTransaction report
     if (!iouRequestActions.length || iouRequestActions.length > 1) {
         return;
     }
 
+    const singleAction = iouRequestActions[0];
+    const originalMessage = getOriginalMessage(singleAction);
+
     // If there's only one IOU request action associated with the report but it's been deleted, then we don't consider this a oneTransaction report
     // and want to display it using the standard view
-    if (isMoneyRequestAction(iouRequestActions[0]) && (getOriginalMessage(iouRequestActions[0])?.deleted ?? '') !== '') {
+    if (isMoneyRequestAction(singleAction) && (originalMessage?.deleted ?? '') !== '') {
```

Comment:
you can check of swapping these 2 can help - missing the `.deleted` can result in bailing out even earlier

Reply:
same here